src/content-steering-controller.js

```diff
@@ -109,57 +110,95 @@ export default class ContentSteeringController extends videojs.EventTarget {
-      this.queryBeforeStart = steeringTag.queryBeforeStart || false;
-      this.proxyServerUrl_ = steeringTag.proxyServerURL;
-    }
+    this.queryBeforeStart = steeringTag.queryBeforeStart || false;
+    this.proxyServerUrl_ = steeringTag.proxyServerURL || null;
 
     // trigger a steering event if we have a pathway from the content steering tag.
     // this tells VHS which segment pathway to start with.
-    if (this.defaultPathway) {
+    // If queryBeforeStart is true we need to wait for the steering manifest response.
+    if (this.defaultPathway && !this.queryBeforeStart) {
       this.trigger('content-steering');
     }
+
+    if (this.queryBeforeStart) {
+      this.requestSteeringManifest(this.steeringManifest.reloadUri);
+    }
   }
 
   /**
    * Requests the content steering manifest and parse the response. This should only be called after
    * assignTagProperties was called with a content steering tag.
+   *
+   * @param {string} initialUri The optional uri to make the request with.
+   *    If set, the request should be made with exactly what is passed in this variable.
+   *    This scenario is specific to DASH when the queryBeforeStart parameter is true.
+   *    This scenario should only happen once on initalization.
    */
-  requestSteeringManifest() {
-    // add parameters to the steering uri
+  requestSteeringManifest(initialUri) {
     const reloadUri = this.steeringManifest.reloadUri;
+
+    if (!initialUri && !reloadUri) {
+      return;
+    }
+
     // We currently don't support passing MPD query parameters directly to the content steering URL as this requires
     // ExtUrlQueryInfo tag support. See the DASH content steering spec section 8.1.
-    const uri = this.proxyServerUrl_ ? this.setProxyServerUrl_(reloadUri) : this.setSteeringParams_(reloadUri);
+
+    // This request URI accounts for manifest URIs that have been excluded.
+    const uri = initialUri || this.getRequestURI(reloadUri);
+
+    // If there are no valid manifest URIs, we should stop content steering.
+    if (!uri) {
+      this.logger_('No valid content steering manifest URIs. Stopping content steering.');
+      this.dispose();
+      this.trigger('error');
+      return;
+    }
 
     this.request_ = this.mainSegmentLoader_.vhs_.xhr({
       uri
-    }, (error) => {
-      // TODO: HLS CASES THAT NEED ADDRESSED:
-      // If the client receives HTTP 410 Gone in response to a manifest request,
-      // it MUST NOT issue another request for that URI for the remainder of the
-      // playback session. It MAY continue to use the most-recently obtained set
-      // of Pathways.
-      // If the client receives HTTP 429 Too Many Requests with a Retry-After
-      // header in response to a manifest request, it SHOULD wait until the time
-      // specified by the Retry-After header to reissue the request.
+    }, (error, errorInfo) => {
       if (error) {
-        // TODO: HLS RETRY CASE:
+        this.requestError_ = error;
```

Comment:
Do we use it somewhere?

Reply:
This was meant so that if someone wanted to see the most recent error at any time, they could use this property on the player. However, it may make more sense to rename it `this.requestError` or create a getter, what do you think?

Reply:
Less code is better. I believe we should keep it only in case we already have a place to use it..

Reply:
Fair enough, as we are logging that error anyways. Removed that in the most recent commit.